write/src/lib.rs

```diff
@@ -37,12 +36,12 @@ pub trait uWrite {
     }
 }
 
-#[cfg(feature = "std")]
-impl uWrite for String {
+impl<T: core::fmt::Write> uWrite for T
+{
     type Error = Infallible;
 
     fn write_str(&mut self, s: &str) -> Result<(), Infallible> {
-        self.push_str(s);
+        self.write_str(s).ok();
```

Comment:
This should not silently ignore the error.

Reply:
Why not? That's what the default impl for `String` did, too. Kinds of makes sense to me.

What alternatives would you have in mind?

Reply:
`type Error = core::fmt::Error;`

Reply:
Fair enough. I've put the alternative implementation into https://github.com/japaric/ufmt/pull/19 .

I'm expecting that one to potentially create overhead the avoidance of which is the whole point of this crate.